index.bs

```diff
@@ -402,8 +403,10 @@ partial interface ModelBuilder {
             interpreted according to the value of *options.layout* and *options.groups*.
         - *options*: an optional {{Conv2dOptions}}. The optional parameters of the operation.
             - *padding*: a sequence of {{long}} of length 4. The padding for the beginning and ending along each spatial dimension of *input*, [beginning_height, ending_height, beginning_width, ending_width]. If not present, the values are assumed to be [0,0,0,0].
+            - *outputPadding*: a sequence of {{long}} of length 2. The padding values applied to each spatial dimension of the output tensor when *options.transpose* is set to true. This explicit padding values are needed to disambiguate the output tensor shape for transposed convolution when the value of the *options.strides* is greater than 1. Note that these values are only used to disambiguate output shape when needed; it does not necessarily cause any padding value to be written to the output tensor. If not specified, the values are assumed to be [0,0]. 
             - *strides*: a sequence of {{long}} of length 2. The stride of the sliding window for each spatial dimension of *input*, [stride_height, stride_width]. If not present, the values are assumed to be [1,1].
             - *dilations*: a sequence of {{long}} of length 2. The dilation factor for each spatial dimension of *input*, [dilation_height, dilation_width]. If not present, the values are assumed to be [1,1].
+            - *transpose*: a {{boolean}} indicating that a transposed convolution operation is performed. Transposed convolution is used in upsampling networks to increase the resolution of a feature as opposed to the typical convolution process that reduces the feature's resolution. When transposed convolution is performed, *options.outputPadding* may be needed to disambiguate the output tensor shape. If not present, this option is assumed to be false.
```

Comment:
I am not sure whether the transposed convolution needs to allow specifying the output shape explicitly. For example, [`conv2d_transpose`](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d_transpose) of TensorFlow uses `output_shape` and [`ConvTranspose`](https://github.com/onnx/onnx/blob/master/docs/Operators.md#ConvTranspose) supports both `output_padding` and `output_shape`. It would require the input size when calculating the output padding from output shape. However, the input size might not be known at the graph build time.

Reply:
We only need either the output paddings or the output shape and never both. This is true for TensorFlow, PyTorch, and ONNX. The output paddings are values specified by the model's author and not inferred from the input shape. And since the output shape can be fully inferred at runtime for the majority case of non-transposed convolutions, I'm more inclined to define only the output padding values as an optional param needed only for transposed convolutions with stride values greater than 1.

Reply:
> We only need either the output paddings or the output shape and never both.

That's correct.

> And since the output shape can be fully inferred at runtime for the majority case of non-transposed convolutions,

I mean the output shape of transposed convolution.

For example, in the ONNX port of fast-transfer [models](https://github.com/microsoft/Windows-Machine-Learning/tree/master/Samples/FNSCandyStyleTransfer/UWP/cs/Assets), the ConvTranspose op sets `output_shape` attribute, e.g. [722, 722]. If we only support output paddings, developers might have to calculate by themselves when building the graph. And according the formula `output padding = output size - ((input size - 1) * stride + filter size - beginning padding - ending padding)`, the input size might need to be inferred from previous layers.

Reply:
I added `outputSizes` as an alternative to `outputPadding` for transposed convolution as suggested here. Please take a look.

Reply:
looks good now, thanks!